subsystem/kubeapi_test.go

```diff
@@ -1821,6 +1823,64 @@ spec:
+		deployAgentClusterInstallCRD(ctx, kubeClient, aciSpec, clusterDeploymentSpec.ClusterInstallRef.Name)
+		clusterKey := types.NamespacedName{
+			Namespace: Options.Namespace,
+			Name:      clusterDeploymentSpec.ClusterName,
+		}
+
+		By("Update cluster's updated_at attribute to become eligible for deregistration due to inactivity")
+		cluster := getClusterFromDB(ctx, kubeClient, db, clusterKey, waitForReconcileTimeout)
+		db.Model(&cluster).UpdateColumn("updated_at", time.Now().AddDate(-1, 0, 0))
+
+		By("Verify no event for deregistering the cluster was emitted")
+		msg := "Cluster is deregistered"
+		Consistently(func() []*common.Event {
+			cluster = getClusterFromDB(ctx, kubeClient, db, clusterKey, 1)
+			eventsHandler := events.New(db, logrus.New())
+			events, err := eventsHandler.GetEvents(*cluster.ID, nil)
+			Expect(err).NotTo(HaveOccurred())
+			return events
+		}, "15s", "5s").ShouldNot(ContainElement(eventMatcher{event: &common.Event{Event: models.Event{Message: &msg, ClusterID: cluster.ID}}}))
+	})
+
+	It("Verify garbage collector deletes unregistered cluster for clusters created via ClusterDeployment", func() {
+		deployClusterDeploymentCRD(ctx, kubeClient, clusterDeploymentSpec)
+		deployInfraEnvCRD(ctx, kubeClient, infraNsName.Name, infraEnvSpec)
+		deployAgentClusterInstallCRD(ctx, kubeClient, aciSpec, clusterDeploymentSpec.ClusterInstallRef.Name)
+		clusterKey := types.NamespacedName{
+			Namespace: Options.Namespace,
+			Name:      clusterDeploymentSpec.ClusterName,
+		}
+		installkey := types.NamespacedName{
+			Namespace: Options.Namespace,
+			Name:      clusterDeploymentSpec.ClusterInstallRef.Name,
+		}
+		cluster := getClusterFromDB(ctx, kubeClient, db, clusterKey, waitForReconcileTimeout)
+
+		By("Delete clusterdeployment to deregister the cluster")
+		Expect(kubeClient.Delete(ctx, getAgentClusterInstallCRD(ctx, kubeClient, installkey))).ShouldNot(HaveOccurred())
+		Expect(kubeClient.Delete(ctx, getClusterDeploymentCRD(ctx, kubeClient, clusterKey))).ShouldNot(HaveOccurred())
+
+		By("Verify cluster record remains in the DB")
+		Consistently(func() error {
+			_, err := common.GetClusterFromDBWhere(db, common.UseEagerLoading, common.IncludeDeletedRecords, "kube_key_name = ? and kube_key_namespace = ?", clusterKey.Name, clusterKey.Namespace)
+			return err
+		}, "15s", "5s").ShouldNot(HaveOccurred())
+
+		By("Update cluster's deleted_at attribute to become eligible for permanent removal")
+		db.Unscoped().Where("id = ?", cluster.ID.String()).Model(&cluster).UpdateColumn("deleted_at", time.Now().AddDate(-1, 0, 0))
+
+		By("Fetch cluster to make sure it was permanently removed by the garbage collector")
+		Eventually(func() error {
+			_, err := common.GetClusterFromDBWhere(db, common.UseEagerLoading, common.IncludeDeletedRecords, "kube_key_name = ? and kube_key_namespace = ?", clusterKey.Name, clusterKey.Namespace)
+			return err
+		}, "30s", "10s").Should(HaveOccurred())
```

Comment:
suggesting to increase it to a minute to be on the safe side.

Reply:
Will do.